server/server_test.go

```diff
@@ -1273,6 +1273,39 @@ var _ = Describe("Server package test", func() {
 		})
 	})
 
+	Describe("Substring Search ", func() {
+		BeforeEach(func() {
+			testURL("POST", networkPluralURL, adminTokenID, getNetwork("red", "red"), http.StatusCreated)
+			testURL("POST", networkPluralURL, adminTokenID, getNetwork("red1", "red"), http.StatusCreated)
+			testURL("POST", networkPluralURL, adminTokenID, getNetwork("red2", "blue"), http.StatusCreated)
+			testURL("POST", networkPluralURL, adminTokenID, getNetwork("red3", "blue"), http.StatusCreated)
+		})
+
+		AfterEach(func() {
+			testURL("DELETE", getNetworkSingularURL("red"), adminTokenID, nil, http.StatusNoContent)
+			testURL("DELETE", getNetworkSingularURL("red1"), adminTokenID, nil, http.StatusNoContent)
+			testURL("DELETE", getNetworkSingularURL("red2"), adminTokenID, nil, http.StatusNoContent)
+			testURL("DELETE", getNetworkSingularURL("red3"), adminTokenID, nil, http.StatusNoContent)
+		})
+
+		DescribeTable("match all rows by substring search", func(query string) {
+			result := testURL("GET", networkPluralURL+query, adminTokenID, nil, http.StatusOK)
+			Expect(result).To(HaveKeyWithValue("networks", ConsistOf(
+				util.MatchAsJSON(getNetwork("red", "red")),
+				util.MatchAsJSON(getNetwork("red1", "red")),
+				util.MatchAsJSON(getNetwork("red2", "blue")),
+				util.MatchAsJSON(getNetwork("red3", "blue")))))
+		},
+			Entry("empty query", ""),
+			Entry("long prefix", "?id=networkred&search_field=id"),
+			Entry("substring at the start", "?id=net&search_field=id"),
+			Entry("substring at the middle", "?id=work&search_field=id"),
+			Entry("substring at the last", "?id=red&search_field=id"),
+			Entry("search on one parameter", "?description=red&search_field=description"),
+			Entry("more than one field to search","?id=red&search_field=id&description=red&search_field=description"),
+		)
+	})
```

Comment:
Can you also add some tests to show that we perform exact matches by default?

Reply:
`search_field=` does not really explain what it does.

Reply:
@cierpuchaw  exact matches would be the get default get queries ...wouldn't existing test case suffice for that... 

search_field..i could change it to search_on_field or search_column